src/loaders/tvg/tvgTvgLoadParser.cpp

```diff
@@ -20,76 +20,85 @@
-    _read_tvg_ui32(&block.length, pointer + TVG_INDICATOR_SIZE);
-    block.data = pointer + TVG_INDICATOR_SIZE + BYTE_COUNTER_SIZE;
-    block.blockEnd = block.data + block.length;
+    block.type = *ptr;
+    _read_tvg_ui32(&block.length, ptr + TVG_INDICATOR_SIZE);
+    block.data = ptr + TVG_INDICATOR_SIZE + BYTE_COUNTER_SIZE;
+    block.end = block.data + block.length;
     return block;
 }
 
-static bool _readTvgHeader(const char **pointer)
+static bool _readTvgHeader(const char **ptr)
 {
-    // Sign phase, always "TVG" declared in TVG_HEADER_TVG_SIGN_CODE
-    if (memcmp(*pointer, TVG_HEADER_TVG_SIGN_CODE, TVG_HEADER_TVG_SIGN_CODE_LENGTH)) return false;
-    *pointer += TVG_HEADER_TVG_SIGN_CODE_LENGTH; // move after sing code
+    //Sign phase, always TVG_BIN_HEADER_SIGNATURE is declared
+    if (memcmp(*ptr, TVG_BIN_HEADER_SIGNATURE, TVG_BIN_HEADER_SIGNATURE_LENGTH)) return false;
+    *ptr += TVG_BIN_HEADER_SIGNATURE_LENGTH;
 
-    // Standard version number, declared in TVG_HEADER_TVG_VERSION_CODE
-    if (memcmp(*pointer, TVG_HEADER_TVG_VERSION_CODE, TVG_HEADER_TVG_VERSION_CODE_LENGTH)) return false;
-    *pointer += TVG_HEADER_TVG_VERSION_CODE_LENGTH; // move after version code
+    //Version number, declared in TVG_BIN_HEADER_VERSION
+    if (memcmp(*ptr, TVG_BIN_HEADER_VERSION, TVG_BIN_HEADER_VERSION_LENGTH)) return false;
+    *ptr += TVG_BIN_HEADER_VERSION_LENGTH;
 
-    // Matadata phase
-    uint16_t meta_length; // Matadata phase length
-    _read_tvg_ui16(&meta_length, *pointer);
-    *pointer += 2; // move after length
+    //Mata data for proof?
+    uint16_t metaLen;
+    _read_tvg_ui16(&metaLen, *ptr);
+    *ptr += 2;
 
+
+//Meta data... Necessary?
 #ifdef THORVG_LOG_ENABLED
-    char metadata[meta_length + 1];
-    memcpy(metadata, *pointer, meta_length);
-    metadata[meta_length] = '\0';
-    printf("TVG_LOADER: Header is valid, metadata[%d]: %s.\n", meta_length, metadata);
+    char metadata[metaLen + 1];
+    memcpy(metadata, *ptr, metaLen);
+    metadata[metaLen] = '\0';
+    printf("TVG_LOADER: Header is valid, metadata[%d]: %s.\n", metaLen, metadata);
 #endif
 
-    *pointer += meta_length;
+    *ptr += metaLen;
+
     return true;
 }
 
-// Paint
-static LoaderResult _parseCmpTarget(const char *pointer, const char *end, Paint *paint)
+
+static LoaderResult _parseCmpTarget(const char *ptr, const char *end, Paint *paint)
 {
-    auto block = _readTvgBlock(pointer);
-    if (block.blockEnd > end) return LoaderResult::SizeCorruption;
+    auto block = _readTvgBlock(ptr);
+    if (block.end > end) return LoaderResult::SizeCorruption;
 
     CompositeMethod cmpMethod;
     if (block.type != TVG_PAINT_CMP_METHOD_INDICATOR) return LoaderResult::LogicalCorruption;
     if (block.length != sizeof(TvgFlag)) return LoaderResult::SizeCorruption;
-    switch (*block.data)
-    {
-        case TVG_PAINT_CMP_METHOD_CLIPPATH_FLAG:
+    switch (*block.data) {
+        case TVG_PAINT_CMP_METHOD_CLIPPATH_FLAG: {
```

Comment:
I know there was a discussion about that... I did casting of the `CompositeMethod` on the `TvgFlag` to avoid switch-case.
If you prefer to do it this way, I can change that in the saver

Reply:
I did like that with join and cap as well.

Reply:
@mgrudzinska hmm this just changes the coding convention. nothing else. if you think you can make it optimal. please do it.

Reply:
I need to change the flags values in tvg_bin_desc file (01, 02, 03 -> 00, 01, 02), so the loader and saver are compatible.